File: accounts/accounts-api/src/main/java/com/hrrm/famoney/accounts/api/MovementDTO.java
package com.hrrm.famoney.accounts.api;

import com.hrrm.famoney.commons.immutables.ImmutableStyle;
import com.hrrm.famoney.jaxrs.DTO;
import com.hrrm.famoney.jaxrs.IdDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;

import java.math.BigDecimal;

@Schema(name = "Movement")
@Value.Immutable
@ImmutableStyle
public interface MovementDTO extends DTO, IdDTO {

    @NotNull
    MovementDataDTO getData();

    @Schema(required = true)
    @NotNull
    Integer getPosition();

    @Schema(required = true)
    @NotNull
    BigDecimal getTotal();

}

File: accounts/accounts-api/src/main/kotlin/com/hrrm/famoney/accounts/api/MovementDTO.kt
package com.hrrm.famoney.accounts.api

import com.hrrm.famoney.jaxrs.DTO
import com.hrrm.famoney.jaxrs.IdDTO
import io.swagger.v3.oas.annotations.media.Schema
import java.math.BigDecimal

@Schema(name = "Movement")
data class MovementDTO(
    override val id: Int,
    val data: MovementDataDTO,
    @get:Schema(required = true)
    val position: Int,
    @get:Schema(required = true)
    val total: BigDecimal,
) : DTO, IdDTO
